projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -4065,27 +4065,33 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
      * @hidden @internal
      */
     public beginAddRowByIndex(rowID: any, index: number, asChild?: boolean) {
+        if (!this.rowEditable) {
+            return;
```

Comment:
Could you please add additional warning message saying that the grid must have rowEditable set to true in order to add new row